### DISCUSSION TITLE
Implementing required status checks with branch protection using webhooks?

### DISCUSSION THREAD

**Discussion by soerenkornetzki** [community/community#24455] [Apps, API and Webhooks]
<p>For one project of mine, I want to implement a required status check as used with the branch protection feature. Unfortunately, none of the GitHub Apps (free or paid) are suitable for my needs, so I need to implement my own required status check.</p>
<p>Is it possible to implement a required status check using GitHub’s Webhooks? I cannot use GitHub Apps as they are hosted at GitHub (and I do not want to disclose private keys etc.) as I understand the environment.</p>
<p>From my understanding, GitHub Webhooks are only informational, and do not affect/bias GitHub’s process. Am I right? How to implement an own/private required status check?</p>

**Comment by helaili** [comment, marked as answer]
<p>A service triggered by a webhook can use the status API and send back a status. Having said that, GitHub Apps can be hosted anywhere, they are actually not hosted by GitHub.</p>
